q2_feature_classifier/classifier.py

```diff
@@ -172,11 +175,33 @@ def _autodetect_orientation(reads, classifier, n=100,
     return chain(reversed_n_reads, (r.reverse_complement() for r in reads))
 
 
-def classify_sklearn(reads: DNAIterator, classifier: Pipeline,
-                     reads_per_batch: int=262144, n_jobs: int=1,
+def _autotune_reads_per_batch(reads, n_jobs):
+    # detect effective jobs. Will raise error if n_jobs == 0
+    n_jobs = sklearn.externals.joblib.effective_n_jobs(n_jobs)
+    # we really only want to calculate this if running in parallel
+    if n_jobs != 1:
+        seq_count = subprocess.run(
+            ['grep', '-c', '^>', str(reads)], check=True,
+            stdout=subprocess.PIPE)
+        return int(int(seq_count.stdout.decode('utf-8')) / n_jobs)
```

Comment:
I don't think is computation is exactly right as it can underestimate the number of reads per batch. 

For example, if the sequence count is 11, and `n_jobs` = 5, then the reads per batch as a float will be 2.2. Casting that to `int` will give 2 reads per batch, so I think ultimately 6 jobs would need to be run. That could really slow things down, I think (since I'm guessing it would wait for one job to complete before starting up a new one?). 

You could get around this by calling `ceil` before `int`, as that would overestimate the number of reads per job. For example:

```python
>>> import numpy as np

>>> int(11/5)
2

>>> int(np.ceil(11/5))
3
```